### PR TITLE
Fixing cred revocation bug where schema was not being saved

### PR DESCRIPTION
**Pull request by nitro-neal**
# Overview
Fixing cred revocation bug

# Description
There was an issue where multiple status list credentials were being created for the same <issuer,schema> pair.
This was because schema was not being saved when the status list credential was created or updated

# How Has This Been Tested?

- [ ✅ ] New unit tests
- [✅ ] Integration Test works
- [✅ ] Manual POSTMAN testing

Debug Output from manual testing:

```
Create Credential:
{
    "credential": {
        "@context": [
            "https://www.w3.org/2018/credentials/v1"
        ],
        "id": "e140cd99-5a42-44b0-b29a-d162dfcbfa41",
        "type": [
            "VerifiableCredential"
        ],
        "issuer": "did:key:z6MkvML6owUEvwUr8P7qsYANnzTBANDq1bBRKdpXoDvTA6sq",
        "issuanceDate": "2023-01-30T21:42:31Z",
        "expirationDate": "2051-10-05T14:48:00.000Z",
        "credentialStatus": {
            "id": "http://localhost:8080/v1/credentials/e140cd99-5a42-44b0-b29a-d162dfcbfa41/status",
            "statusListCredential": "http://localhost:8080/v1/credentials/status/56bf85e4-82ed-49d6-804d-9abf957c34d2",
            "statusListIndex": "93914",
            "statusPurpose": "revocation",
            "type": "StatusList2021Entry"
        },
        "credentialSubject": {
            "employedAt": "2022-08-20T13:20:10.000+0000",
            "givenName": "Alice",
            "id": "did:key:z6MkvML6owUEvwUr8P7qsYANnzTBANDq1bBRKdpXoDvTA6sq"
        },
        "credentialSchema": {
            "id": "d3501080-7f2f-4b63-952c-c243b6f1bdb8",
            "type": "JsonSchemaValidator2018"
        }
    },
    "credentialJwt": "eyJhbGciOiJFZERTQSIsImtpZCI6ImRpZDprZXk6ejZNa3ZNTDZvd1VFdndVcjhQN3FzWUFObnpUQkFORHExYkJSS2RwWG9EdlRBNnNxIiwidHlwIjoiSldUIn0.eyJleHAiOjI1ODAxMzAwODAsImlzcyI6ImRpZDprZXk6ejZNa3ZNTDZvd1VFdndVcjhQN3FzWUFObnpUQkFORHExYkJSS2RwWG9EdlRBNnNxIiwianRpIjoiZTE0MGNkOTktNWE0Mi00NGIwLWIyOWEtZDE2MmRmY2JmYTQxIiwibmJmIjoxNjc1MTE0OTUxLCJzdWIiOiJkaWQ6a2V5Ono2TWt2TUw2b3dVRXZ3VXI4UDdxc1lBTm56VEJBTkRxMWJCUktkcFhvRHZUQTZzcSIsInZjIjp7IkBjb250ZXh0IjpbImh0dHBzOi8vd3d3LnczLm9yZy8yMDE4L2NyZWRlbnRpYWxzL3YxIl0sImlkIjoiZTE0MGNkOTktNWE0Mi00NGIwLWIyOWEtZDE2MmRmY2JmYTQxIiwidHlwZSI6WyJWZXJpZmlhYmxlQ3JlZGVudGlhbCJdLCJpc3N1ZXIiOiJkaWQ6a2V5Ono2TWt2TUw2b3dVRXZ3VXI4UDdxc1lBTm56VEJBTkRxMWJCUktkcFhvRHZUQTZzcSIsImlzc3VhbmNlRGF0ZSI6IjIwMjMtMDEtMzBUMjE6NDI6MzFaIiwiZXhwaXJhdGlvbkRhdGUiOiIyMDUxLTEwLTA1VDE0OjQ4OjAwLjAwMFoiLCJjcmVkZW50aWFsU3RhdHVzIjp7ImlkIjoiaHR0cDovL2xvY2FsaG9zdDo4MDgwL3YxL2NyZWRlbnRpYWxzL2UxNDBjZDk5LTVhNDItNDRiMC1iMjlhLWQxNjJkZmNiZmE0MS9zdGF0dXMiLCJzdGF0dXNMaXN0Q3JlZGVudGlhbCI6Imh0dHA6Ly9sb2NhbGhvc3Q6ODA4MC92MS9jcmVkZW50aWFscy9zdGF0dXMvNTZiZjg1ZTQtODJlZC00OWQ2LTgwNGQtOWFiZjk1N2MzNGQyIiwic3RhdHVzTGlzdEluZGV4IjoiOTM5MTQiLCJzdGF0dXNQdXJwb3NlIjoicmV2b2NhdGlvbiIsInR5cGUiOiJTdGF0dXNMaXN0MjAyMUVudHJ5In0sImNyZWRlbnRpYWxTdWJqZWN0Ijp7ImVtcGxveWVkQXQiOiIyMDIyLTA4LTIwVDEzOjIwOjEwLjAwMCswMDAwIiwiZ2l2ZW5OYW1lIjoiQWxpY2UiLCJpZCI6ImRpZDprZXk6ejZNa3ZNTDZvd1VFdndVcjhQN3FzWUFObnpUQkFORHExYkJSS2RwWG9EdlRBNnNxIn0sImNyZWRlbnRpYWxTY2hlbWEiOnsiaWQiOiJkMzUwMTA4MC03ZjJmLTRiNjMtOTUyYy1jMjQzYjZmMWJkYjgiLCJ0eXBlIjoiSnNvblNjaGVtYVZhbGlkYXRvcjIwMTgifX19.YeJoWjiY2u0oWiJsTvg2vU4QnpsJc4jgA0tgGaRT0o7tE_TEAeEnTIipGghtCJx1NvLG9Vw9P-I7imNcOJ7ABg"
}



Credential Status: - http://localhost:8080/v1/credentials/e140cd99-5a42-44b0-b29a-d162dfcbfa41/status
{"revoked":false}



Status List Credential - http://localhost:8080/v1/credentials/status/56bf85e4-82ed-49d6-804d-9abf957c34d2
{
   "id":"http://localhost:8080/v1/credentials/status/56bf85e4-82ed-49d6-804d-9abf957c34d2",
   "credential":{
      "@context":[
         "https://www.w3.org/2018/credentials/v1",
         "https://w3id.org/vc/status-list/2021/v1"
      ],
      "id":"http://localhost:8080/v1/credentials/status/56bf85e4-82ed-49d6-804d-9abf957c34d2",
      "type":[
         "VerifiableCredential",
         "StatusList2021Credential"
      ],
      "issuer":"did:key:z6MkvML6owUEvwUr8P7qsYANnzTBANDq1bBRKdpXoDvTA6sq",
      "issuanceDate":"2023-01-30T21:42:31Z",
      "credentialSubject":{
         "encodedList":"H4sIAAAAAAAA/2IAAweGUTAKRsEoGAWjYBSMPAAIAAD//9BoYmEICAAA",
         "id":"http://localhost:8080/v1/credentials/status/56bf85e4-82ed-49d6-804d-9abf957c34d2",
         "statusPurpose":"revocation",
         "type":"StatusList2021"
      },
      "credentialSchema":{
         "id":"d3501080-7f2f-4b63-952c-c243b6f1bdb8",
         "type":"JsonSchemaValidator2018"
      }
   },
   "credentialJwt":"eyJhbGciOiJFZERTQSIsImtpZCI6ImRpZDprZXk6ejZNa3ZNTDZvd1VFdndVcjhQN3FzWUFObnpUQkFORHExYkJSS2RwWG9EdlRBNnNxIiwidHlwIjoiSldUIn0.eyJpc3MiOiJkaWQ6a2V5Ono2TWt2TUw2b3dVRXZ3VXI4UDdxc1lBTm56VEJBTkRxMWJCUktkcFhvRHZUQTZzcSIsImp0aSI6Imh0dHA6Ly9sb2NhbGhvc3Q6ODA4MC92MS9jcmVkZW50aWFscy9zdGF0dXMvNTZiZjg1ZTQtODJlZC00OWQ2LTgwNGQtOWFiZjk1N2MzNGQyIiwibmJmIjoxNjc1MTE0OTUxLCJzdWIiOiJodHRwOi8vbG9jYWxob3N0OjgwODAvdjEvY3JlZGVudGlhbHMvc3RhdHVzLzU2YmY4NWU0LTgyZWQtNDlkNi04MDRkLTlhYmY5NTdjMzRkMiIsInZjIjp7IkBjb250ZXh0IjpbImh0dHBzOi8vd3d3LnczLm9yZy8yMDE4L2NyZWRlbnRpYWxzL3YxIiwiaHR0cHM6Ly93M2lkLm9yZy92Yy9zdGF0dXMtbGlzdC8yMDIxL3YxIl0sImlkIjoiaHR0cDovL2xvY2FsaG9zdDo4MDgwL3YxL2NyZWRlbnRpYWxzL3N0YXR1cy81NmJmODVlNC04MmVkLTQ5ZDYtODA0ZC05YWJmOTU3YzM0ZDIiLCJ0eXBlIjpbIlZlcmlmaWFibGVDcmVkZW50aWFsIiwiU3RhdHVzTGlzdDIwMjFDcmVkZW50aWFsIl0sImlzc3VlciI6ImRpZDprZXk6ejZNa3ZNTDZvd1VFdndVcjhQN3FzWUFObnpUQkFORHExYkJSS2RwWG9EdlRBNnNxIiwiaXNzdWFuY2VEYXRlIjoiMjAyMy0wMS0zMFQyMTo0MjozMVoiLCJjcmVkZW50aWFsU3ViamVjdCI6eyJlbmNvZGVkTGlzdCI6Ikg0c0lBQUFBQUFBQS8ySUFBd2VHVVRBS1JzRW9HQVdqWUJTTVBBQUlBQUQvLzlCb1ltRUlDQUFBIiwiaWQiOiJodHRwOi8vbG9jYWxob3N0OjgwODAvdjEvY3JlZGVudGlhbHMvc3RhdHVzLzU2YmY4NWU0LTgyZWQtNDlkNi04MDRkLTlhYmY5NTdjMzRkMiIsInN0YXR1c1B1cnBvc2UiOiJyZXZvY2F0aW9uIiwidHlwZSI6IlN0YXR1c0xpc3QyMDIxIn0sImNyZWRlbnRpYWxTY2hlbWEiOnsiaWQiOiJkMzUwMTA4MC03ZjJmLTRiNjMtOTUyYy1jMjQzYjZmMWJkYjgiLCJ0eXBlIjoiSnNvblNjaGVtYVZhbGlkYXRvcjIwMTgifX19.pTB7bLQoFZnO_JZmyDDropiX9uG5znrj1YvVarQH8ShcjTlYqUPE9c7wIM1dWejsjoL6TXqluXplPVSdESEcAg"
}


PUT - http://localhost:8080/v1/credentials/e140cd99-5a42-44b0-b29a-d162dfcbfa41/status
{
    "revoked": true
}


Status List Credential (updated encoded list) - http://localhost:8080/v1/credentials/status/56bf85e4-82ed-49d6-804d-9abf957c34d2
{
   "id":"http://localhost:8080/v1/credentials/status/56bf85e4-82ed-49d6-804d-9abf957c34d2",
   "credential":{
      "@context":[
         "https://www.w3.org/2018/credentials/v1",
         "https://w3id.org/vc/status-list/2021/v1"
      ],
      "id":"http://localhost:8080/v1/credentials/status/56bf85e4-82ed-49d6-804d-9abf957c34d2",
      "type":[
         "VerifiableCredential",
         "StatusList2021Credential"
      ],
      "issuer":"did:key:z6MkvML6owUEvwUr8P7qsYANnzTBANDq1bBRKdpXoDvTA6sq",
      "issuanceDate":"2023-01-30T21:47:39Z",
      "credentialSubject":{
         "encodedList":"H4sIAAAAAAAA/+zAQQkAIBRAsSeY18KWEVP8y9a3zg0AAAAAAAAAAAAAgHG7egEAAP//FE5O++gtAAA=",
         "id":"http://localhost:8080/v1/credentials/status/56bf85e4-82ed-49d6-804d-9abf957c34d2",
         "statusPurpose":"revocation",
         "type":"StatusList2021"
      },
      "credentialSchema":{
         "id":"d3501080-7f2f-4b63-952c-c243b6f1bdb8",
         "type":"JsonSchemaValidator2018"
      }
   },
   "credentialJwt":"eyJhbGciOiJFZERTQSIsImtpZCI6ImRpZDprZXk6ejZNa3ZNTDZvd1VFdndVcjhQN3FzWUFObnpUQkFORHExYkJSS2RwWG9EdlRBNnNxIiwidHlwIjoiSldUIn0.eyJpc3MiOiJkaWQ6a2V5Ono2TWt2TUw2b3dVRXZ3VXI4UDdxc1lBTm56VEJBTkRxMWJCUktkcFhvRHZUQTZzcSIsImp0aSI6Imh0dHA6Ly9sb2NhbGhvc3Q6ODA4MC92MS9jcmVkZW50aWFscy9zdGF0dXMvNTZiZjg1ZTQtODJlZC00OWQ2LTgwNGQtOWFiZjk1N2MzNGQyIiwibmJmIjoxNjc1MTE1MjU5LCJzdWIiOiJodHRwOi8vbG9jYWxob3N0OjgwODAvdjEvY3JlZGVudGlhbHMvc3RhdHVzLzU2YmY4NWU0LTgyZWQtNDlkNi04MDRkLTlhYmY5NTdjMzRkMiIsInZjIjp7IkBjb250ZXh0IjpbImh0dHBzOi8vd3d3LnczLm9yZy8yMDE4L2NyZWRlbnRpYWxzL3YxIiwiaHR0cHM6Ly93M2lkLm9yZy92Yy9zdGF0dXMtbGlzdC8yMDIxL3YxIl0sImlkIjoiaHR0cDovL2xvY2FsaG9zdDo4MDgwL3YxL2NyZWRlbnRpYWxzL3N0YXR1cy81NmJmODVlNC04MmVkLTQ5ZDYtODA0ZC05YWJmOTU3YzM0ZDIiLCJ0eXBlIjpbIlZlcmlmaWFibGVDcmVkZW50aWFsIiwiU3RhdHVzTGlzdDIwMjFDcmVkZW50aWFsIl0sImlzc3VlciI6ImRpZDprZXk6ejZNa3ZNTDZvd1VFdndVcjhQN3FzWUFObnpUQkFORHExYkJSS2RwWG9EdlRBNnNxIiwiaXNzdWFuY2VEYXRlIjoiMjAyMy0wMS0zMFQyMTo0NzozOVoiLCJjcmVkZW50aWFsU3ViamVjdCI6eyJlbmNvZGVkTGlzdCI6Ikg0c0lBQUFBQUFBQS8rekFRUWtBSUJSQXNTZVkxOEtXRVZQOHk5YTN6ZzBBQUFBQUFBQUFBQUFBZ0hHN2VnRUFBUC8vRkU1TysrZ3RBQUE9IiwiaWQiOiJodHRwOi8vbG9jYWxob3N0OjgwODAvdjEvY3JlZGVudGlhbHMvc3RhdHVzLzU2YmY4NWU0LTgyZWQtNDlkNi04MDRkLTlhYmY5NTdjMzRkMiIsInN0YXR1c1B1cnBvc2UiOiJyZXZvY2F0aW9uIiwidHlwZSI6IlN0YXR1c0xpc3QyMDIxIn0sImNyZWRlbnRpYWxTY2hlbWEiOnsiaWQiOiJkMzUwMTA4MC03ZjJmLTRiNjMtOTUyYy1jMjQzYjZmMWJkYjgiLCJ0eXBlIjoiSnNvblNjaGVtYVZhbGlkYXRvcjIwMTgifX19.ljbYI9h6zLJaWwJY__VrHjnQzVnRsQLq5HV4LlkyIxmcXyz60iyk1o_teS2FjmoSqvtZQO26iWE1vaAg_v1ODA"
}

Create Credential (same issuer and schema, same statusListCredential):
{
    "credential": {
        "@context": [
            "https://www.w3.org/2018/credentials/v1"
        ],
        "id": "8b7b2130-3d56-4ab0-a6e3-ab0e5bcf6161",
        "type": [
            "VerifiableCredential"
        ],
        "issuer": "did:key:z6MkvML6owUEvwUr8P7qsYANnzTBANDq1bBRKdpXoDvTA6sq",
        "issuanceDate": "2023-01-30T21:49:24Z",
        "expirationDate": "2051-10-05T14:48:00.000Z",
        "credentialStatus": {
            "id": "http://localhost:8080/v1/credentials/8b7b2130-3d56-4ab0-a6e3-ab0e5bcf6161/status",
            "statusListCredential": "http://localhost:8080/v1/credentials/status/56bf85e4-82ed-49d6-804d-9abf957c34d2",
            "statusListIndex": "125650",
            "statusPurpose": "revocation",
            "type": "StatusList2021Entry"
        },
        "credentialSubject": {
            "employedAt": "2022-08-20T13:20:10.000+0000",
            "givenName": "Alice",
            "id": "did:key:z6MkvML6owUEvwUr8P7qsYANnzTBANDq1bBRKdpXoDvTA6sq"
        },
        "credentialSchema": {
            "id": "d3501080-7f2f-4b63-952c-c243b6f1bdb8",
            "type": "JsonSchemaValidator2018"
        }
    },
    "credentialJwt": "eyJhbGciOiJFZERTQSIsImtpZCI6ImRpZDprZXk6ejZNa3ZNTDZvd1VFdndVcjhQN3FzWUFObnpUQkFORHExYkJSS2RwWG9EdlRBNnNxIiwidHlwIjoiSldUIn0.eyJleHAiOjI1ODAxMzAwODAsImlzcyI6ImRpZDprZXk6ejZNa3ZNTDZvd1VFdndVcjhQN3FzWUFObnpUQkFORHExYkJSS2RwWG9EdlRBNnNxIiwianRpIjoiOGI3YjIxMzAtM2Q1Ni00YWIwLWE2ZTMtYWIwZTViY2Y2MTYxIiwibmJmIjoxNjc1MTE1MzY0LCJzdWIiOiJkaWQ6a2V5Ono2TWt2TUw2b3dVRXZ3VXI4UDdxc1lBTm56VEJBTkRxMWJCUktkcFhvRHZUQTZzcSIsInZjIjp7IkBjb250ZXh0IjpbImh0dHBzOi8vd3d3LnczLm9yZy8yMDE4L2NyZWRlbnRpYWxzL3YxIl0sImlkIjoiOGI3YjIxMzAtM2Q1Ni00YWIwLWE2ZTMtYWIwZTViY2Y2MTYxIiwidHlwZSI6WyJWZXJpZmlhYmxlQ3JlZGVudGlhbCJdLCJpc3N1ZXIiOiJkaWQ6a2V5Ono2TWt2TUw2b3dVRXZ3VXI4UDdxc1lBTm56VEJBTkRxMWJCUktkcFhvRHZUQTZzcSIsImlzc3VhbmNlRGF0ZSI6IjIwMjMtMDEtMzBUMjE6NDk6MjRaIiwiZXhwaXJhdGlvbkRhdGUiOiIyMDUxLTEwLTA1VDE0OjQ4OjAwLjAwMFoiLCJjcmVkZW50aWFsU3RhdHVzIjp7ImlkIjoiaHR0cDovL2xvY2FsaG9zdDo4MDgwL3YxL2NyZWRlbnRpYWxzLzhiN2IyMTMwLTNkNTYtNGFiMC1hNmUzLWFiMGU1YmNmNjE2MS9zdGF0dXMiLCJzdGF0dXNMaXN0Q3JlZGVudGlhbCI6Imh0dHA6Ly9sb2NhbGhvc3Q6ODA4MC92MS9jcmVkZW50aWFscy9zdGF0dXMvNTZiZjg1ZTQtODJlZC00OWQ2LTgwNGQtOWFiZjk1N2MzNGQyIiwic3RhdHVzTGlzdEluZGV4IjoiMTI1NjUwIiwic3RhdHVzUHVycG9zZSI6InJldm9jYXRpb24iLCJ0eXBlIjoiU3RhdHVzTGlzdDIwMjFFbnRyeSJ9LCJjcmVkZW50aWFsU3ViamVjdCI6eyJlbXBsb3llZEF0IjoiMjAyMi0wOC0yMFQxMzoyMDoxMC4wMDArMDAwMCIsImdpdmVuTmFtZSI6IkFsaWNlIiwiaWQiOiJkaWQ6a2V5Ono2TWt2TUw2b3dVRXZ3VXI4UDdxc1lBTm56VEJBTkRxMWJCUktkcFhvRHZUQTZzcSJ9LCJjcmVkZW50aWFsU2NoZW1hIjp7ImlkIjoiZDM1MDEwODAtN2YyZi00YjYzLTk1MmMtYzI0M2I2ZjFiZGI4IiwidHlwZSI6Ikpzb25TY2hlbWFWYWxpZGF0b3IyMDE4In19fQ.m8kAkVtSAAe4vib0_YlndiTSsZ-tGMhCAJcA10Fd56kz1c6JbZeoDqs_lRncYP1cDnO_hTTdZQ6W-N6fwdeYDg"
}

Encoded List didn't Change
"encodedList":"H4sIAAAAAAAA/+zAQQkAIBRAsSeY18KWEVP8y9a3zg0AAAAAAAAAAAAAgHG7egEAAP//FE5O++gtAAA=",

Revoked 2nd cred
Status list changes - H4sIAAAAAAAA/+zOQQ0AIBADsJGcf3O4QAVBxR60CppnnR0AAAAAAAAAAAAAAOqmHQAAAIAfTHIDAAD//yY2kr5oPQAA


Create Credential(new schema)

{
    "credential": {
        "@context": [
            "https://www.w3.org/2018/credentials/v1"
        ],
        "id": "52a0d4bb-64af-46e4-b1ca-a7ba7940e9b2",
        "type": [
            "VerifiableCredential"
        ],
        "issuer": "did:key:z6MkvML6owUEvwUr8P7qsYANnzTBANDq1bBRKdpXoDvTA6sq",
        "issuanceDate": "2023-01-30T21:51:31Z",
        "expirationDate": "2051-10-05T14:48:00.000Z",
        "credentialStatus": {
            "id": "http://localhost:8080/v1/credentials/52a0d4bb-64af-46e4-b1ca-a7ba7940e9b2/status",
            "statusListCredential": "http://localhost:8080/v1/credentials/status/a5bc3d7f-d850-46fd-a4b8-8f65adb13ba2",
            "statusListIndex": "59681",
            "statusPurpose": "revocation",
            "type": "StatusList2021Entry"
        },
        "credentialSubject": {
            "employedAt": "2022-08-20T13:20:10.000+0000",
            "givenName": "Alice",
            "id": "did:key:z6MkvML6owUEvwUr8P7qsYANnzTBANDq1bBRKdpXoDvTA6sq"
        },
        "credentialSchema": {
            "id": "83cf658e-85a8-444d-8363-9822c135ac42",
            "type": "JsonSchemaValidator2018"
        }
    },
    "credentialJwt": "eyJhbGciOiJFZERTQSIsImtpZCI6ImRpZDprZXk6ejZNa3ZNTDZvd1VFdndVcjhQN3FzWUFObnpUQkFORHExYkJSS2RwWG9EdlRBNnNxIiwidHlwIjoiSldUIn0.eyJleHAiOjI1ODAxMzAwODAsImlzcyI6ImRpZDprZXk6ejZNa3ZNTDZvd1VFdndVcjhQN3FzWUFObnpUQkFORHExYkJSS2RwWG9EdlRBNnNxIiwianRpIjoiNTJhMGQ0YmItNjRhZi00NmU0LWIxY2EtYTdiYTc5NDBlOWIyIiwibmJmIjoxNjc1MTE1NDkxLCJzdWIiOiJkaWQ6a2V5Ono2TWt2TUw2b3dVRXZ3VXI4UDdxc1lBTm56VEJBTkRxMWJCUktkcFhvRHZUQTZzcSIsInZjIjp7IkBjb250ZXh0IjpbImh0dHBzOi8vd3d3LnczLm9yZy8yMDE4L2NyZWRlbnRpYWxzL3YxIl0sImlkIjoiNTJhMGQ0YmItNjRhZi00NmU0LWIxY2EtYTdiYTc5NDBlOWIyIiwidHlwZSI6WyJWZXJpZmlhYmxlQ3JlZGVudGlhbCJdLCJpc3N1ZXIiOiJkaWQ6a2V5Ono2TWt2TUw2b3dVRXZ3VXI4UDdxc1lBTm56VEJBTkRxMWJCUktkcFhvRHZUQTZzcSIsImlzc3VhbmNlRGF0ZSI6IjIwMjMtMDEtMzBUMjE6NTE6MzFaIiwiZXhwaXJhdGlvbkRhdGUiOiIyMDUxLTEwLTA1VDE0OjQ4OjAwLjAwMFoiLCJjcmVkZW50aWFsU3RhdHVzIjp7ImlkIjoiaHR0cDovL2xvY2FsaG9zdDo4MDgwL3YxL2NyZWRlbnRpYWxzLzUyYTBkNGJiLTY0YWYtNDZlNC1iMWNhLWE3YmE3OTQwZTliMi9zdGF0dXMiLCJzdGF0dXNMaXN0Q3JlZGVudGlhbCI6Imh0dHA6Ly9sb2NhbGhvc3Q6ODA4MC92MS9jcmVkZW50aWFscy9zdGF0dXMvYTViYzNkN2YtZDg1MC00NmZkLWE0YjgtOGY2NWFkYjEzYmEyIiwic3RhdHVzTGlzdEluZGV4IjoiNTk2ODEiLCJzdGF0dXNQdXJwb3NlIjoicmV2b2NhdGlvbiIsInR5cGUiOiJTdGF0dXNMaXN0MjAyMUVudHJ5In0sImNyZWRlbnRpYWxTdWJqZWN0Ijp7ImVtcGxveWVkQXQiOiIyMDIyLTA4LTIwVDEzOjIwOjEwLjAwMCswMDAwIiwiZ2l2ZW5OYW1lIjoiQWxpY2UiLCJpZCI6ImRpZDprZXk6ejZNa3ZNTDZvd1VFdndVcjhQN3FzWUFObnpUQkFORHExYkJSS2RwWG9EdlRBNnNxIn0sImNyZWRlbnRpYWxTY2hlbWEiOnsiaWQiOiI4M2NmNjU4ZS04NWE4LTQ0NGQtODM2My05ODIyYzEzNWFjNDIiLCJ0eXBlIjoiSnNvblNjaGVtYVZhbGlkYXRvcjIwMTgifX19.-o9pU9ZAxMmw6Rso6rLwhCxMpkdgimMWrbEQffQ7xg8Yd-EuxYYR0RPncvYhIDjIsNgAWA8Pv_Jo-sgldJIgCQ"
}


Different statusCredentialList! :check:
```

### CODE REVIEW

**Review by decentralgabe — APPROVED** [review]
is there a way to test this across DB impls we have?

File: pkg/server/router/credential_test.go [hunk 1]
```diff
@@ -274,5 +275,132 @@ func TestCredentialRouter(t *testing.T) {
...
+
+		credStatusListID := strings.Split(credStatusListStr, "/")[len(strings.Split(credStatusListStr, "/"))-1]
+		credStatusList, err := credService.GetCredentialStatusList(context.Background(), credential.GetCredentialStatusListRequest{ID: credStatusListID})
+		assert.NoError(tt, err)
+		assert.Equal(tt, credStatusList.Credential.ID, credStatusMap["statusListCredential"].(string))
+
+		credentialSubject := credStatusList.Container.Credential.CredentialSubject
+		assert.NotEmpty(tt, credentialSubject)
+
+		encodedList := credentialSubject["encodedList"]
```
Comment (andresuribe87) [inline pkg/server/router/credential_test.go:387]: Would it make sense to assert that all elements of encodedList are zero?

File: pkg/server/router/credential_test.go [hunk 2]
```diff
@@ -274,5 +275,132 @@ func TestCredentialRouter(t *testing.T) {
...
+		assert.NoError(tt, err)
+		assert.Equal(tt, credStatusListAfterRevoke.Credential.ID, credStatusMap["statusListCredential"].(string))
+
+		credentialSubjectAfterRevoke := credStatusListAfterRevoke.Container.Credential.CredentialSubject
+		assert.NotEmpty(tt, credentialSubjectAfterRevoke)
+
+		encodedListAfterRevoke := credentialSubjectAfterRevoke["encodedList"]
+		assert.NotEmpty(tt, encodedListAfterRevoke)
+
+		assert.NotEqualValues(tt, encodedListAfterRevoke, encodedList)
```
Comment (andresuribe87) [inline pkg/server/router/credential_test.go:404]: Similar to above, would it make sense to assert that only a single element of `encodedListAfterRevoke` is equal to 1?

**Review by andresuribe87 — APPROVED** [review]
Small recs

File: pkg/server/router/credential_test.go (hunk 2 above)
Comment (nitro-neal) [reply pkg/server/router/credential_test.go:404]: yup I think there will need to be work done in the SDK, I will get that functionality up and running

File: pkg/server/router/credential_test.go [hunk 3]
```diff
@@ -274,5 +277,256 @@ func TestCredentialRouter(t *testing.T) {
...
+		keyStoreService := testKeyStoreService(tt, bolt)
+		didService := testDIDService(tt, bolt, keyStoreService)
+		schemaService := testSchemaService(tt, bolt, keyStoreService, didService)
+		credService, err := credential.NewCredentialService(serviceConfig, bolt, keyStoreService, didService.GetResolver(), schemaService)
+		assert.NoError(tt, err)
+		assert.NotEmpty(tt, credService)
+
+		// check type and status
+		assert.Equal(tt, framework.Credential, credService.Type())
+		assert.Equal(tt, framework.StatusReady, credService.Status().Status)
```
Comment (andresuribe87) [inline pkg/server/router/credential_test.go:327]: Why is this needed?

File: pkg/server/router/credential_test.go [hunk 4]
```diff
@@ -274,5 +277,256 @@ func TestCredentialRouter(t *testing.T) {
...
+
+		assert.Contains(tt, credStatusMapThree["id"], fmt.Sprintf("v1/credentials/%s/status", createdCredThree.ID))
+		assert.Contains(tt, credStatusMapThree["statusListCredential"], "v1/credentials/status")
+		assert.NotEmpty(tt, credStatusMapThree["statusListIndex"])
+
+		// Cred with different <issuer, schema> pair have different statusListCredential
+		assert.NotEqual(tt, credStatusMapThree["statusListCredential"], credStatusMap["statusListCredential"])
+	})
+
+	t.Run("Credential Status List Test No Schemas", func(tt *testing.T) {
```
Comment (andresuribe87) [inline pkg/server/router/credential_test.go:313]: What behavior is being tested here? I took a stab at it. Even trying to do it, I couldn't figure out why there were 3 credentials being created in the test.

See [here](https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#naming-your-tests) for best practices around naming things. Maybe the test could be split into multiple tests. 

```suggestion
	t.Run("Creating Revocable credentials with different schemas have different status list", func(tt *testing.T) {
```

File: pkg/server/router/credential_test.go [hunk 5]
```diff
@@ -274,5 +277,256 @@ func TestCredentialRouter(t *testing.T) {
...
+
+		updatedStatus, err := credService.UpdateCredentialStatus(context.Background(), credential.UpdateCredentialStatusRequest{ID: createdCred.ID, Revoked: true})
+		assert.NoError(tt, err)
+		assert.Equal(tt, updatedStatus.Revoked, true)
+
+		credStatusListAfterRevoke, err := credService.GetCredentialStatusList(context.Background(), credential.GetCredentialStatusListRequest{ID: credStatusListID})
+		assert.NoError(tt, err)
+		assert.Equal(tt, credStatusListAfterRevoke.Credential.ID, statusEntry.StatusListCredential)
+
+		// Validate the StatusListIndex in flipped in the credStatusList
```
Comment (andresuribe87) [inline pkg/server/router/credential_test.go:519]:
```suggestion
		// Validate the StatusListIndex is flipped in the credStatusList
```

File: pkg/server/router/credential_test.go [hunk 6]
```diff
@@ -274,5 +277,256 @@ func TestCredentialRouter(t *testing.T) {
...
+		valid, err = status.ValidateCredentialInStatusList(*createdCred.Credential, *credStatusListAfterRevoke.Credential)
```
Comment (andresuribe87) [inline pkg/server/router/credential_test.go:520]: Separate point, this function name is not very informative. I always have to end up looking what `Validate` means in this context.
